validation/validate.py

```diff
@@ -0,0 +1,268 @@
+import json
+import math
+import docker
+import numpy as np
+from argparse import ArgumentParser, Namespace
+from sklearn.metrics import accuracy_score, recall_score, precision_score
+from sklearn.metrics.pairwise import paired_distances
+from sklearn.model_selection import KFold
+from os.path import join, exists
+from enum import Enum
+from typing import List, Union, Tuple, cast
+
+Match = Tuple[str, int, int]
+Mismatch = Tuple[str, int, str, int]
+Pair = Union[Match, Mismatch]
+
+Path = Tuple[str, str]
+Label = bool
+
+
+class DistanceMetric(Enum):
```

Comment:
What does this enum buy us over using a [choices argument type](https://docs.python.org/3/library/argparse.html#choices) in Argparse and mapping the resulting known-string directly to a distance function reference which then gets passed around the script?

Reply:
I was unaware of choices, thanks for bringing it to my attention! I have changed the names of the distance metrics to reflect what they are (squared euclidean distance and angular distance). I am giving them default int values, instead of cosine/euclidean to denote that they are different than the sklearn metrics and to be more accurate to the implementation. Choices is really helpful though, and I am including it in addition to the enum!